### ISSUE TITLE
Unable to turn rules off in `next` branch

### ISSUE THREAD

**Issue by bmish** [ember-template-lint/ember-template-lint#1644]
I tried using the `next` branch (for the upcoming V3 release #1315) and it seems that my configuration file now fails to turn off rules, here's an example:

This fails to turn the eol-last rule off (violations are reported from this rule still):

```js
// .template-lintrc.js
module.exports = {
  extends: ['stylistic'],
  rules: {
    'eol-last': false,
  },
};
```

Note that this successfully turns the eol-last rule on:

```js
// .template-lintrc.js
module.exports = {
  extends: [],
  rules: {
    'eol-last': ['error', 'editorconfig'],
  },
};
```

Note that the behavior is correct on master.

**Comment by bmish** [comment]
@Turbo87 I verified that this was caused by your change in #1513 - "Fix `extends` overriding order"

**Comment by Turbo87** [comment]
😱 did I screw up? 😞

**Comment by rwjblue** [comment]
One thing to point out here is that the correct way to turn it off would be:

```js
// .template-lintrc.js
module.exports = {
  extends: ['stylistic'],
  rules: {
    'eol-last': 'off',
  },
};
```

(Note: `'off'` vs `false`)

**Comment by rwjblue** [comment]
@bmish - Can you make a failing test case branch/PR?
